### DISCUSSION TITLE
Querying Photos by Creating App

### DISCUSSION THREAD

**Discussion by djbeadle** [RhetTbull/osxphotos#936] [Q&A] [answered]
On iOS and in the Photos app I can search for the app that created the photo. Is there any way to query this after the export?

This software is amazing, by the way. I have 53,186 assets in my library and this is giving me a way to liberate them 👏

![image](https://user-images.githubusercontent.com/6235378/213584286-cd6ff484-3012-4002-a23c-275ca2bd9200.png)

**Comment by RhetTbull** [comment]
Hi. It's not currently possible to do this in Photos but after you asked, I checked the database and I think I know how to pull this data out (at least on Ventura--I don't think the data is present on earlier versions of macOS).  I'll open a new issue to add this feature.

**Comment by oPromessa** [comment]
Would this be the same tag as "EXIF:Software"?

```sh
$ exiftool -G -u -a -software P5290141.JPG 
[EXIF]          Software                        : Adobe Photoshop Album 2.0
```

**Reply by RhetTbull** [reply]
> Would this be the same tag as "EXIF:Software"?

No unfortunately it's not. the `EXIF:Software` tag depends on the app to set it. Some do, for example Instagram, but others, like Messages, do not. I've decoded this though and will push it out shortly in new release. There's a new attribute: `PhotoInfo.search_info.source` that lists this

**Comment by RhetTbull** [comment, marked as answer]
This is now include in [v0.56.4](https://github.com/RhetTbull/osxphotos/releases/tag/v0.56.4).  You can filter photos by source with a `--query-eval`:

`osxphotos query --query-eval "photo.search_info.source == 'Safari'"`
